scripts/build/Jenkins/Jenkinsfile

```diff
@@ -415,6 +415,18 @@ def ExportTestResults(Map options, String platform, String type, String workspac
     }
 }
 
+def ExportTestScreenshots(Map options, String workspace, String platformName, String jobName, Map params) {
+    catchError(message: "Error exporting test screenshots (this won't fail the build)", buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+        def screenshotsFolder = '${workspace}/${ENGINE_REPOSITORY_NAME}/AutomatedTesting/user/PythonTests/Automated/Screenshots'
+        def s3Uploader = '${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/build/tools/upload_to_s3.py'
+        def command = '${options.PYTHON_DIR}/python.cmd -u ${s3Uploader} --base_dir ${screenshotsFolder} ' +
+                      '--file_regex "(.*png$|.*ppm$)" --bucket ${env.TEST_SCREENSHOT_BUCKET} '
+                      '--search_subdirectories True --key_prefix ${branchName}_${env.BUILD_NUMBER}'
+        bat label: "Uploading test screenshots for ${jobName}",
+        script: command
```

Comment:
it may be worth to delete the screenshots after uploading, not sure if its possible that these screenshots will remain between consecutive runs, as the image seems to be reused for incremental build

Reply:
For the duplicate screenshots concern, the `--key_prefix ${branchName}_${env.BUILD_NUMBER}` should deal with that on s3 because it will label each one by build number. 
I'll look into the zip/tar idea as well as deleting screenshots locally after uploading but want to sync with infra over it first as I am not sure how much of a concern it is locally (the tests have been taking and storing screenshots since nightly GPU runs became a thing).
Also, as a side note, these s3 uploads only occur on test failure + if the job has the  `"TEST_SCREENSHOTS": "True"` parameter (which only the nightly `test_gpu_profile_vs2019` job has).